src/client/metadataTransfer.ts

```diff
@@ -191,10 +190,21 @@ export abstract class MetadataTransfer<Status extends MetadataRequestStatus, Res
       completed = true;
       this.canceled = false;
     } else {
-      mdapiStatus = await this.checkStatus();
-      completed = mdapiStatus?.done;
-      if (!completed) {
-        this.event.emit('update', mdapiStatus);
+      try {
+        mdapiStatus = await this.checkStatus();
+        completed = mdapiStatus?.done;
+        if (!completed) {
+          this.event.emit('update', mdapiStatus);
+        }
+      } catch (e) {
+        this.logger.error(e);
+        // tolerate intermittent network errors upto retry limit
+        if (e instanceof Error && (e.message.includes('ETIMEDOUT') || e.message.includes('ENOTFOUND'))) {
```

Comment:
@mshanemc would you want to include connection reset?

examples: ECONNREFUSED|ECONNRESET/, /Error: socket hang up/, /connection reset by peer/

Reply:
> Will this make commands that encounter a network issue return a 0 exit code?
> 
> Timed out deploy/retrieves and push/pulls returning a 0 exit code was a big issue that [FinancialForce](https://gus.lightning.force.com/one/one.app#eyJjb21wb25lbnREZWYiOiJmb3JjZUNoYXR0ZXI6ZGVza3RvcENoYXR0ZXIiLCJhdHRyaWJ1dGVzIjp7ImZlZWRFbGVtZW50SWQiOiIwRDVFRTAwMDAwS2NFTWYwQU4ifSwic3RhdGUiOnt9fQ%3D%3D) was seeing in their CI recently. Like [#1301](https://github.com/forcedotcom/cli/issues/1301) suggests, the default retry could be 0

it'll keep polling...so if the connection NEVER comes back then the polling client would exit and hit the usual 69 code (same as if the mdapi operation didn't complete).

Reply:
Good ideas, Pete.  I see ECONNRESET and `socket hang up` so I'm gonna add those.